```
Error running mint automation
+ mkdir -p 10218-a719eea/gopath/src/github.com/minio
+ git -C 10218-a719eea/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 10218-a719eea/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 10218-a719eea/gopath/src/github.com/minio/minio fetch --quiet minio pull/10218/head:pr10218
+ git -C 10218-a719eea/gopath/src/github.com/minio/minio checkout --quiet pr10218
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/10218-a719eea/gopath
+ make -C 10218-a719eea/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/Azure/azure-pipeline-go v0.2.1
go: downloading google.golang.org/api v0.5.0
go: downloading github.com/minio/sha256-simd v0.1.1
go: downloading github.com/minio/minio-go/v7 v7.0.3
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.3.0
go: downloading github.com/json-iterator/go v1.1.10
go: downloading golang.org/x/crypto v0.0.0-20200709230013-948cd5f35899
go: downloading github.com/shirou/gopsutil v2.20.3-0.20200314133625-53cec6b37e6a+incompatible
go: downloading github.com/klauspost/readahead v1.3.1
go: extracting github.com/Azure/azure-pipeline-go v0.2.1
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: extracting github.com/shirou/gopsutil v2.20.3-0.20200314133625-53cec6b37e6a+incompatible
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: extracting github.com/json-iterator/go v1.1.10
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: extracting github.com/minio/sha256-simd v0.1.1
go: extracting github.com/klauspost/readahead v1.3.1
go: downloading golang.org/x/net v0.0.0-20200707034311-ab3426394381
go: extracting github.com/minio/minio-go/v7 v7.0.3
go: extracting github.com/mitchellh/go-homedir v1.1.0
go: extracting github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: extracting github.com/mattn/go-ieproxy v0.0.1
go: downloading github.com/klauspost/compress v1.10.3
go: downloading github.com/colinmarc/hdfs/v2 v2.1.1
go: downloading github.com/minio/highwayhash v1.0.0
go: downloading github.com/minio/cli v1.22.0
go: downloading go.etcd.io/etcd/v3 v3.3.0-rc.0.0.20200707003333-58bb8ae09f8e
go: extracting golang.org/x/crypto v0.0.0-20200709230013-948cd5f35899
go: extracting github.com/minio/cli v1.22.0
go: downloading github.com/minio/md5-simd v1.1.0
go: extracting github.com/colinmarc/hdfs/v2 v2.1.1
go: downloading github.com/miekg/dns v1.1.8
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: extracting github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: extracting github.com/minio/md5-simd v1.1.0
go: downloading github.com/secure-io/sio-go v0.3.0
go: extracting github.com/miekg/dns v1.1.8
go: extracting github.com/secure-io/sio-go v0.3.0
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: extracting github.com/minio/highwayhash v1.0.0
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: extracting github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/prometheus/client_golang v1.0.0
go: extracting github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading golang.org/x/sys v0.0.0-20200625212154-ddb9806d33ae
go: extracting github.com/prometheus/client_golang v1.0.0
go: downloading gopkg.in/ini.v1 v1.57.0
go: extracting golang.org/x/net v0.0.0-20200707034311-ab3426394381
go: extracting go.etcd.io/etcd/v3 v3.3.0-rc.0.0.20200707003333-58bb8ae09f8e
go: extracting gopkg.in/ini.v1 v1.57.0
go: downloading github.com/gorilla/rpc v1.2.0
go: downloading cloud.google.com/go v0.39.0
go: downloading github.com/golang/protobuf v1.4.0
go: extracting github.com/gorilla/rpc v1.2.0
go: extracting github.com/golang/protobuf v1.4.0
go: downloading github.com/fatih/color v1.7.0
go: extracting golang.org/x/sys v0.0.0-20200625212154-ddb9806d33ae
go: downloading github.com/google/uuid v1.1.1
go: extracting github.com/fatih/color v1.7.0
go: extracting github.com/google/uuid v1.1.1
go: downloading github.com/cheggaaa/pb v1.0.28
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: extracting github.com/cheggaaa/pb v1.0.28
go: downloading github.com/modern-go/reflect2 v1.0.1
go: extracting github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/beorn7/perks v1.0.0
go: extracting github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/Azure/azure-storage-blob-go v0.8.0
go: extracting github.com/Azure/azure-storage-blob-go v0.8.0
go: downloading google.golang.org/grpc v1.26.0
go: extracting github.com/beorn7/perks v1.0.0
go: extracting cloud.google.com/go v0.39.0
go: downloading google.golang.org/protobuf v1.22.0
go: extracting google.golang.org/grpc v1.26.0
go: downloading github.com/mattn/go-colorable v0.1.4
go: extracting github.com/mattn/go-colorable v0.1.4
go: downloading github.com/rs/xid v1.2.1
go: extracting github.com/rs/xid v1.2.1
go: downloading github.com/ncw/directio v1.0.5
go: extracting github.com/ncw/directio v1.0.5
go: downloading gopkg.in/olivere/elastic.v5 v5.0.80
go: extracting google.golang.org/protobuf v1.22.0
go: downloading github.com/gogo/protobuf v1.3.1
go: extracting gopkg.in/olivere/elastic.v5 v5.0.80
go: downloading google.golang.org/genproto v0.0.0-20190819201941-24fa4b261c55
go: extracting github.com/klauspost/compress v1.10.3
go: extracting github.com/gogo/protobuf v1.3.1
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/gorilla/handlers v1.4.2
go: downloading github.com/djherbis/atime v1.0.0
go: extracting github.com/gorilla/handlers v1.4.2
go: extracting github.com/dustin/go-humanize v1.0.0
go: downloading github.com/willf/bloom v2.0.3+incompatible
go: downloading github.com/lib/pq v1.7.0
go: extracting github.com/djherbis/atime v1.0.0
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: extracting github.com/willf/bloom v2.0.3+incompatible
go: downloading github.com/gorilla/mux v1.7.5-0.20200711200521-98cb6bf42e08
go: extracting google.golang.org/api v0.5.0
go: extracting github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/nsqio/go-nsq v1.0.7
go: extracting github.com/lib/pq v1.7.0
go: downloading github.com/minio/simdjson-go v0.1.5-0.20200303142138-b17fe061ea37
go: extracting github.com/nsqio/go-nsq v1.0.7
go: extracting github.com/gorilla/mux v1.7.5-0.20200711200521-98cb6bf42e08
go: downloading github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4
go: downloading github.com/minio/selfupdate v0.3.1
go: extracting github.com/minio/selfupdate v0.3.1
go: downloading github.com/pkg/errors v0.8.1
go: extracting github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4
go: extracting github.com/pkg/errors v0.8.1
go: downloading github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: extracting github.com/minio/simdjson-go v0.1.5-0.20200303142138-b17fe061ea37
go: downloading github.com/nats-io/stan.go v0.7.0
go: extracting github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: downloading github.com/beevik/ntp v0.2.0
go: extracting github.com/nats-io/stan.go v0.7.0
go: downloading go.uber.org/zap v1.14.1
go: extracting github.com/beevik/ntp v0.2.0
go: downloading github.com/prometheus/common v0.4.1
go: extracting go.uber.org/zap v1.14.1
go: extracting github.com/prometheus/common v0.4.1
go: downloading github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: extracting google.golang.org/genproto v0.0.0-20190819201941-24fa4b261c55
go: downloading github.com/rs/cors v1.7.0
go: extracting github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading github.com/nats-io/nuid v1.0.1
go: extracting github.com/rs/cors v1.7.0
go: downloading github.com/coreos/go-semver v0.2.0
go: extracting github.com/nats-io/nuid v1.0.1
go: downloading github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: extracting github.com/coreos/go-semver v0.2.0
go: downloading go.opencensus.io v0.21.0
go: extracting github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: downloading github.com/Shopify/sarama v1.24.1
go: downloading github.com/minio/sio v0.2.0
go: downloading github.com/prometheus/procfs v0.0.8
go: extracting go.opencensus.io v0.21.0
go: downloading golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: extracting github.com/minio/sio v0.2.0
go: downloading gopkg.in/yaml.v2 v2.2.8
go: extracting github.com/Shopify/sarama v1.24.1
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: extracting github.com/prometheus/procfs v0.0.8
go: extracting golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: extracting gopkg.in/yaml.v2 v2.2.8
go: extracting github.com/spaolacci/murmur3 v1.1.0
go: downloading github.com/tinylib/msgp v1.1.2
go: extracting github.com/tinylib/msgp v1.1.2
go: downloading github.com/nats-io/nats.go v1.10.0
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/eapache/go-resiliency v1.1.0
go: extracting github.com/cespare/xxhash/v2 v2.1.1
go: extracting github.com/nats-io/nats.go v1.10.0
go: downloading github.com/rjeczalik/notify v0.9.2
go: extracting github.com/eapache/go-resiliency v1.1.0
go: downloading github.com/klauspost/pgzip v1.2.1
go: downloading github.com/philhofer/fwd v1.0.0
go: extracting github.com/rjeczalik/notify v0.9.2
go: downloading gopkg.in/ldap.v3 v3.0.3
go: extracting github.com/philhofer/fwd v1.0.0
go: extracting github.com/klauspost/pgzip v1.2.1
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/coredns/coredns v1.4.0
go: extracting gopkg.in/ldap.v3 v3.0.3
go: extracting github.com/fatih/structs v1.1.0
go: downloading golang.org/x/text v0.3.3
go: downloading github.com/willf/bitset v1.1.10
go: extracting github.com/willf/bitset v1.1.10
go: downloading github.com/nats-io/nkeys v0.1.4
go: extracting github.com/coredns/coredns v1.4.0
go: extracting github.com/nats-io/nkeys v0.1.4
go: downloading github.com/eapache/queue v1.1.0
go: extracting github.com/eapache/queue v1.1.0
go: downloading github.com/mattn/go-runewidth v0.0.2
go: extracting github.com/mattn/go-runewidth v0.0.2
go: extracting golang.org/x/text v0.3.3
go: extracting gopkg.in/jcmturner/gokrb5.v7 v7.3.0
go: downloading github.com/rcrowley/go-metrics v0.0.0-20181016184325-3113b8401b8a
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: downloading github.com/golang/snappy v0.0.1
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading github.com/nats-io/jwt v0.3.2
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/hashicorp/go-uuid v1.0.1
go: downloading github.com/coreos/go-systemd/v22 v22.0.0
go: extracting github.com/dchest/siphash v1.2.1
go: extracting github.com/rcrowley/go-metrics v0.0.0-20181016184325-3113b8401b8a
go: downloading github.com/klauspost/reedsolomon v1.9.9
go: downloading github.com/klauspost/cpuid v1.3.1
go: extracting github.com/xdg/stringprep v1.0.0
go: extracting github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: extracting github.com/golang/snappy v0.0.1
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/montanaflynn/stats v0.5.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: extracting github.com/hashicorp/go-uuid v1.0.1
go: extracting github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/pierrec/lz4 v2.4.0+incompatible
go: extracting github.com/nats-io/jwt v0.3.2
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading github.com/alecthomas/participle v0.2.1
go: extracting github.com/coreos/go-systemd/v22 v22.0.0
go: downloading git.apache.org/thrift.git v0.13.0
go: extracting github.com/klauspost/reedsolomon v1.9.9
go: downloading go.uber.org/multierr v1.5.0
go: extracting github.com/googleapis/gax-go/v2 v2.0.4
go: downloading github.com/hashicorp/golang-lru v0.5.1
go: extracting github.com/klauspost/cpuid v1.3.1
go: extracting github.com/montanaflynn/stats v0.5.0
go: extracting github.com/go-sql-driver/mysql v1.5.0
go: extracting github.com/davecgh/go-spew v1.1.1
go: extracting github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: extracting go.uber.org/multierr v1.5.0
go: downloading github.com/eclipse/paho.mqtt.golang v1.2.0
go: downloading gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: extracting github.com/alecthomas/participle v0.2.1
go: downloading go.uber.org/atomic v1.6.0
go: downloading github.com/mattn/go-isatty v0.0.8
go: downloading github.com/tidwall/sjson v1.0.4
go: extracting github.com/hashicorp/golang-lru v0.5.1
go: downloading github.com/tidwall/gjson v1.3.5
go: extracting go.uber.org/atomic v1.6.0
go: extracting gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: extracting gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: extracting gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: downloading github.com/hashicorp/vault/api v1.0.4
go: extracting github.com/mattn/go-isatty v0.0.8
go: extracting github.com/tidwall/sjson v1.0.4
go: extracting github.com/eclipse/paho.mqtt.golang v1.2.0
go: extracting github.com/tidwall/gjson v1.3.5
go: downloading github.com/tidwall/match v1.0.1
go: downloading github.com/tidwall/pretty v1.0.0
go: extracting gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: extracting gopkg.in/jcmturner/aescts.v1 v1.0.1
go: extracting github.com/hashicorp/vault/api v1.0.4
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/hashicorp/go-retryablehttp v0.5.4
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading gopkg.in/square/go-jose.v2 v2.3.1
go: downloading github.com/hashicorp/vault/sdk v0.1.13
go: extracting github.com/tidwall/match v1.0.1
go: extracting github.com/tidwall/pretty v1.0.0
go: downloading github.com/hashicorp/hcl v1.0.0
go: extracting github.com/hashicorp/errwrap v1.0.0
go: extracting golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: extracting github.com/hashicorp/go-retryablehttp v0.5.4
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: extracting gopkg.in/square/go-jose.v2 v2.3.1
go: extracting github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: extracting github.com/mitchellh/mapstructure v1.1.2
go: extracting github.com/hashicorp/hcl v1.0.0
go: extracting github.com/hashicorp/vault/sdk v0.1.13
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: extracting github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: downloading github.com/ryanuber/go-glob v1.0.0
go: extracting github.com/hashicorp/go-rootcerts v1.0.1
go: extracting github.com/ryanuber/go-glob v1.0.0
go: extracting github.com/hashicorp/go-sockaddr v1.0.2
go: extracting git.apache.org/thrift.git v0.13.0
go: extracting github.com/pierrec/lz4 v2.4.0+incompatible
go: finding github.com/colinmarc/hdfs/v2 v2.1.1
go: finding github.com/minio/minio-go/v7 v7.0.3
go: finding github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: finding gopkg.in/jcmturner/gokrb5.v7 v7.3.0
go: finding github.com/minio/cli v1.22.0
go: finding github.com/minio/sha256-simd v0.1.1
go: finding github.com/Azure/azure-pipeline-go v0.2.1
go: finding github.com/json-iterator/go v1.1.10
go: finding github.com/Azure/azure-storage-blob-go v0.8.0
go: finding cloud.google.com/go v0.39.0
go: finding github.com/dustin/go-humanize v1.0.0
go: finding github.com/minio/highwayhash v1.0.0
go: finding google.golang.org/api v0.5.0
go: finding github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: finding github.com/hashicorp/go-uuid v1.0.1
go: finding github.com/mattn/go-colorable v0.1.4
go: finding github.com/mattn/go-isatty v0.0.8
go: finding golang.org/x/crypto v0.0.0-20200709230013-948cd5f35899
go: finding gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: finding github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: finding github.com/mitchellh/go-homedir v1.1.0
go: finding github.com/fatih/color v1.7.0
go: finding github.com/montanaflynn/stats v0.5.0
go: finding gopkg.in/ini.v1 v1.57.0
go: finding github.com/gorilla/mux v1.7.5-0.20200711200521-98cb6bf42e08
go: finding github.com/golang/protobuf v1.4.0
go: finding github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: finding github.com/mattn/go-ieproxy v0.0.1
go: finding github.com/rjeczalik/notify v0.9.2
go: finding golang.org/x/sys v0.0.0-20200625212154-ddb9806d33ae
go: finding github.com/modern-go/reflect2 v1.0.1
go: finding golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: finding google.golang.org/genproto v0.0.0-20190819201941-24fa4b261c55
go: finding github.com/googleapis/gax-go/v2 v2.0.4
go: finding google.golang.org/grpc v1.26.0
go: finding github.com/secure-io/sio-go v0.3.0
go: finding go.opencensus.io v0.21.0
go: finding github.com/shirou/gopsutil v2.20.3-0.20200314133625-53cec6b37e6a+incompatible
go: finding github.com/ncw/directio v1.0.5
go: finding github.com/google/uuid v1.1.1
go: finding github.com/minio/md5-simd v1.1.0
go: finding google.golang.org/protobuf v1.22.0
go: finding golang.org/x/net v0.0.0-20200707034311-ab3426394381
go: finding gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: finding github.com/rs/xid v1.2.1
go: finding gopkg.in/jcmturner/aescts.v1 v1.0.1
go: finding github.com/klauspost/cpuid v1.3.1
go: finding golang.org/x/text v0.3.3
go: finding gopkg.in/jcmturner/rpc.v1 v1.1.0
go: finding github.com/cespare/xxhash/v2 v2.1.1
go: finding github.com/cheggaaa/pb v1.0.28
go: finding github.com/hashicorp/golang-lru v0.5.1
go: finding github.com/mattn/go-runewidth v0.0.2
go: finding github.com/dchest/siphash v1.2.1
go: finding github.com/djherbis/atime v1.0.0
go: finding github.com/elazarl/go-bindata-assetfs v1.0.0
go: finding github.com/gorilla/handlers v1.4.2
go: finding github.com/gorilla/rpc v1.2.0
go: finding github.com/klauspost/compress v1.10.3
go: finding github.com/klauspost/readahead v1.3.1
go: finding github.com/klauspost/reedsolomon v1.9.9
go: finding github.com/miekg/dns v1.1.8
go: finding go.etcd.io/etcd/v3 v3.3.0-rc.0.0.20200707003333-58bb8ae09f8e
go: finding github.com/gogo/protobuf v1.3.1
go: finding go.uber.org/zap v1.14.1
go: finding go.uber.org/atomic v1.6.0
go: finding go.uber.org/multierr v1.5.0
go: finding github.com/coreos/go-systemd/v22 v22.0.0
go: finding github.com/coreos/go-semver v0.2.0
go: finding github.com/coredns/coredns v1.4.0
go: finding gopkg.in/ldap.v3 v3.0.3
go: finding gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: finding github.com/Shopify/sarama v1.24.1
go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/eapache/go-resiliency v1.1.0
go: finding github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: finding github.com/golang/snappy v0.0.1
go: finding github.com/eapache/queue v1.1.0
go: finding github.com/pierrec/lz4 v2.4.0+incompatible
go: finding github.com/rcrowley/go-metrics v0.0.0-20181016184325-3113b8401b8a
go: finding github.com/eclipse/paho.mqtt.golang v1.2.0
go: finding github.com/go-sql-driver/mysql v1.5.0
go: finding github.com/gomodule/redigo v2.0.0+incompatible
go: finding github.com/lib/pq v1.7.0
go: finding github.com/nats-io/nats.go v1.10.0
go: finding github.com/nats-io/jwt v0.3.2
go: finding github.com/nats-io/nkeys v0.1.4
go: finding github.com/nats-io/nuid v1.0.1
go: finding github.com/nats-io/stan.go v0.7.0
go: finding github.com/nsqio/go-nsq v1.0.7
go: finding github.com/pkg/errors v0.8.1
go: finding github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: finding github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: finding github.com/xdg/stringprep v1.0.0
go: finding gopkg.in/olivere/elastic.v5 v5.0.80
go: finding github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: finding github.com/hashicorp/vault/api v1.0.4
go: finding github.com/hashicorp/errwrap v1.0.0
go: finding github.com/hashicorp/go-cleanhttp v0.5.1
go: finding github.com/hashicorp/go-multierror v1.0.0
go: finding github.com/hashicorp/go-retryablehttp v0.5.4
go: finding github.com/hashicorp/go-rootcerts v1.0.1
go: finding github.com/hashicorp/hcl v1.0.0
go: finding github.com/hashicorp/vault/sdk v0.1.13
go: finding github.com/hashicorp/go-sockaddr v1.0.2
go: finding github.com/ryanuber/go-glob v1.0.0
go: finding github.com/mitchellh/mapstructure v1.1.2
go: finding golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: finding gopkg.in/square/go-jose.v2 v2.3.1
go: finding github.com/minio/sio v0.2.0
go: finding github.com/beevik/ntp v0.2.0
go: finding github.com/fatih/structs v1.1.0
go: finding gopkg.in/yaml.v2 v2.2.8
go: finding github.com/klauspost/pgzip v1.2.1
go: finding github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: finding github.com/alecthomas/participle v0.2.1
go: finding github.com/minio/simdjson-go v0.1.5-0.20200303142138-b17fe061ea37
go: finding git.apache.org/thrift.git v0.13.0
go: finding github.com/tidwall/gjson v1.3.5
go: finding github.com/tidwall/match v1.0.1
go: finding github.com/tidwall/pretty v1.0.0
go: finding github.com/tidwall/sjson v1.0.4
go: finding github.com/minio/selfupdate v0.3.1
go: finding github.com/prometheus/client_golang v1.0.0
go: finding github.com/beorn7/perks v1.0.0
go: finding github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4
go: finding github.com/prometheus/common v0.4.1
go: finding github.com/matttproud/golang_protobuf_extensions v1.0.1
go: finding github.com/prometheus/procfs v0.0.8
go: finding github.com/rs/cors v1.7.0
go: finding github.com/tinylib/msgp v1.1.2
go: finding github.com/philhofer/fwd v1.0.0
go: finding github.com/willf/bloom v2.0.3+incompatible
go: finding github.com/spaolacci/murmur3 v1.1.0
go: finding github.com/willf/bitset v1.1.10
# github.com/minio/minio/cmd
cmd/background-newdisks-heal-ops.go:66:10: ep declared and not used
make: *** [build] Error 2

```

--- a/cmd/admin-handlers.go
+++ b/cmd/admin-handlers.go
@@ -799,6 +799,52 @@ func (a adminAPIHandlers) HealHandler(w http.ResponseWriter, r *http.Request) {
 	keepConnLive(w, r, respCh)
 }
 
+func getAggregatedBackgroundHealState(ctx context.Context, failOnErr bool) (madmin.BgHealState, error) {
+	var bgHealStates []madmin.BgHealState
+
+	// Get local heal status first
+	bgHealStates = append(bgHealStates, getLocalBackgroundHealStatus())
+
+	if globalIsDistErasure {
+		// Get heal status from other peers
+		peersHealStates, nerrs := globalNotificationSys.BackgroundHealStatus()
+		for _, nerr := range nerrs {
+			if nerr.Err != nil {
+				if failOnErr {
+					return madmin.BgHealState{}, nerr.Err
+				}
+				logger.LogIf(ctx, nerr.Err)
+			}
+		}
+		bgHealStates = append(bgHealStates, peersHealStates...)
+	}
+
+	// Aggregate healing result
+	var aggregatedHealStateResult = madmin.BgHealState{
+		ScannedItemsCount: bgHealStates[0].ScannedItemsCount,
+		LastHealActivity:  bgHealStates[0].LastHealActivity,
+		NextHealRound:     bgHealStates[0].NextHealRound,
+		HealDisks:         bgHealStates[0].HealDisks,
+	}
+
+	bgHealStates = bgHealStates[1:]
+
+	for _, state := range bgHealStates {
+		aggregatedHealStateResult.ScannedItemsCount += state.ScannedItemsCount
+		aggregatedHealStateResult.HealDisks = append(aggregatedHealStateResult.HealDisks, state.HealDisks...)
+		if !state.LastHealActivity.IsZero() && aggregatedHealStateResult.LastHealActivity.Before(state.LastHealActivity) {
+			aggregatedHealStateResult.LastHealActivity = state.LastHealActivity
+			// The node which has the last heal activity means its
+			// is the node that is orchestrating self healing operations,
+			// which also means it is the same node which decides when
+			// the next self healing operation will be done.
+			aggregatedHealStateResult.NextHealRound = state.NextHealRound
+		}
+	}
+
+	return aggregatedHealStateResult, nil
+}
+
 func (a adminAPIHandlers) BackgroundHealStatusHandler(w http.ResponseWriter, r *http.Request) {
 	ctx := newContext(r, w, "HealBackgroundStatus")
 
@@ -815,39 +861,8 @@ func (a adminAPIHandlers) BackgroundHealStatusHandler(w http.ResponseWriter, r *
 		return
 	}
 
-	var bgHealStates []madmin.BgHealState
-
-	// Get local heal status first
-	bgHealStates = append(bgHealStates, getLocalBackgroundHealStatus())
-
-	if globalIsDistErasure {
-		// Get heal status from other peers
-		peersHealStates := globalNotificationSys.BackgroundHealStatus()
-		bgHealStates = append(bgHealStates, peersHealStates...)
-	}
-
-	// Aggregate healing result
-	var aggregatedHealStateResult = madmin.BgHealState{
-		ScannedItemsCount: bgHealStates[0].ScannedItemsCount,
-		LastHealActivity:  bgHealStates[0].LastHealActivity,
-		NextHealRound:     bgHealStates[0].NextHealRound,
-	}
-
-	bgHealStates = bgHealStates[1:]
-
-	for _, state := range bgHealStates {
-		aggregatedHealStateResult.ScannedItemsCount += state.ScannedItemsCount
-		if !state.LastHealActivity.IsZero() && aggregatedHealStateResult.LastHealActivity.Before(state.LastHealActivity) {
-			aggregatedHealStateResult.LastHealActivity = state.LastHealActivity
-			// The node which has the last heal activity means its
-			// is the node that is orchestrating self healing operations,
-			// which also means it is the same node which decides when
-			// the next self healing operation will be done.
-			aggregatedHealStateResult.NextHealRound = state.NextHealRound
-		}
-	}
-
-	if err := json.NewEncoder(w).Encode(aggregatedHealStateResult); err != nil {
+	aggregateHealStateResult, _ := getAggregatedBackgroundHealState(r.Context(), false)
+	if err := json.NewEncoder(w).Encode(aggregateHealStateResult); err != nil {
 		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
 		return
 	}

--- a/cmd/admin-heal-ops.go
+++ b/cmd/admin-heal-ops.go
@@ -88,7 +88,8 @@ type allHealState struct {
 	sync.Mutex
 
 	// map of heal path to heal sequence
-	healSeqMap map[string]*healSequence
+	healSeqMap     map[string]*healSequence
+	healLocalDisks []Endpoints
 }
 
 // newHealState - initialize global heal state management
@@ -100,6 +101,22 @@ func newHealState() *allHealState {
 	go healState.periodicHealSeqsClean(GlobalContext)
 
 	return healState
+}
+
+func (ahs *allHealState) getHealLocalDisks() []Endpoints {
+	ahs.Lock()
+	defer ahs.Unlock()
+
+	healLocalDisks := make([]Endpoints, len(ahs.healLocalDisks))
+	copy(healLocalDisks, ahs.healLocalDisks)
+	return healLocalDisks
+}
+
+func (ahs *allHealState) updateHealLocalDisks(eps []Endpoints) {
+	ahs.Lock()
+	defer ahs.Unlock()
+
+	ahs.healLocalDisks = eps
 }
 
 func (ahs *allHealState) periodicHealSeqsClean(ctx context.Context) {

--- a/cmd/background-heal-ops.go
+++ b/cmd/background-heal-ops.go
@@ -90,6 +90,9 @@ func (h *healRoutine) run(ctx context.Context, objAPI ObjectLayer) {
 			case task.bucket == nopHeal:
 				continue
 			case task.bucket == SlashSeparator:
+				// Quickly check if drives need healing upon start-up
+				globalBackgroundHealState.updateHealLocalDisks(getLocalDisksToHeal(objAPI))
+
 				res, err = healDiskFormat(ctx, objAPI, task.opts)
 			case task.bucket != "" && task.object == "":
 				res, err = objAPI.HealBucket(ctx, task.bucket, task.opts.DryRun, task.opts.Remove)

--- a/cmd/background-newdisks-heal-ops.go
+++ b/cmd/background-newdisks-heal-ops.go
@@ -18,6 +18,7 @@ package cmd
 
 import (
 	"context"
+	"fmt"
 	"time"
 
 	"github.com/dustin/go-humanize"
@@ -28,6 +29,46 @@ const defaultMonitorNewDiskInterval = time.Minute * 3
 
 func initLocalDisksAutoHeal(ctx context.Context, objAPI ObjectLayer) {
 	go monitorLocalDisksAndHeal(ctx, objAPI)
+}
+
+func getLocalDisksToHeal(objAPI ObjectLayer) []Endpoints {
+	z, ok := objAPI.(*erasureZones)
+	if !ok {
+		return nil
+	}
+
+	// Attempt a heal as the server starts-up first.
+	localDisksInZoneHeal := make([]Endpoints, len(z.zones))
+	for i, ep := range globalEndpoints {
+		localDisksToHeal := Endpoints{}
+		for _, endpoint := range ep.Endpoints {
+			if !endpoint.IsLocal {
+				continue
+			}
+			// Try to connect to the current endpoint
+			// and reformat if the current disk is not formatted
+			_, _, err := connectEndpoint(endpoint)
+			if err == errUnformattedDisk {
+				localDisksToHeal = append(localDisksToHeal, endpoint)
+			}
+		}
+		if len(localDisksToHeal) == 0 {
+			continue
+		}
+		localDisksInZoneHeal[i] = localDisksToHeal
+	}
+	return localDisksInZoneHeal
+
+}
+
+func getDrivesToHealCount(localDisksInZoneHeal []Endpoints) int {
+	var drivesToHeal int
+	for _, eps := range localDisksInZoneHeal {
+		for range eps {
+			drivesToHeal++
+		}
+	}
+	return drivesToHeal
 }
 
 // monitorLocalDisksAndHeal - ensures that detected new disks are healed
@@ -50,52 +91,42 @@ func monitorLocalDisksAndHeal(ctx context.Context, objAPI ObjectLayer) {
 		time.Sleep(time.Second)
 	}
 
+	localDisksInZoneHeal := globalBackgroundHealState.getHealLocalDisks()
+
+	drivesToHeal := getDrivesToHealCount(localDisksInZoneHeal)
+	if drivesToHeal != 0 {
+		logger.Info(fmt.Sprintf("Found drives to heal %d, waiting until %s to heal the content...",
+			drivesToHeal, defaultMonitorNewDiskInterval))
+	}
+
+	firstTime := true
+
 	// Perform automatic disk healing when a disk is replaced locally.
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case <-time.After(defaultMonitorNewDiskInterval):
-			// Attempt a heal as the server starts-up first.
-			localDisksInZoneHeal := make([]Endpoints, len(z.zones))
-			var healNewDisks bool
-			for i, ep := range globalEndpoints {
-				localDisksToHeal := Endpoints{}
-				for _, endpoint := range ep.Endpoints {
-					if !endpoint.IsLocal {
-						continue
-					}
-					// Try to connect to the current endpoint
-					// and reformat if the current disk is not formatted
-					_, _, err := connectEndpoint(endpoint)
-					if err == errUnformattedDisk {
-						localDisksToHeal = append(localDisksToHeal, endpoint)
-					}
-				}
-				if len(localDisksToHeal) == 0 {
+			// heal only if new disks found.
+			if drivesToHeal == 0 {
+				firstTime = false
+				localDisksInZoneHeal = getLocalDisksToHeal(z)
+				drivesToHeal = getDrivesToHealCount(localDisksInZoneHeal)
+				if drivesToHeal == 0 {
+					// No drives to heal.
+					globalBackgroundHealState.updateHealLocalDisks(nil)
 					continue
 				}
-				localDisksInZoneHeal[i] = localDisksToHeal
-				healNewDisks = true
+				globalBackgroundHealState.updateHealLocalDisks(localDisksInZoneHeal)
 			}
 
-			// Reformat disks only if needed.
-			if !healNewDisks {
-				continue
+			if !firstTime {
+				// Reformat disks
+				bgSeq.sourceCh <- healSource{bucket: SlashSeparator}
+
+				// Ensure that reformatting disks is finished
+				bgSeq.sourceCh <- healSource{bucket: nopHeal}
 			}
-
-			logger.Info("New unformatted drives detected attempting to heal...")
-			for i, disks := range localDisksInZoneHeal {
-				for _, disk := range disks {
-					logger.Info("Healing disk '%s' on %s zone", disk, humanize.Ordinal(i+1))
-				}
-			}
-
-			// Reformat disks
-			bgSeq.sourceCh <- healSource{bucket: SlashSeparator}
-
-			// Ensure that reformatting disks is finished
-			bgSeq.sourceCh <- healSource{bucket: nopHeal}
 
 			var erasureSetInZoneToHeal = make([][]int, len(localDisksInZoneHeal))
 			// Compute the list of erasure set to heal
@@ -108,6 +139,7 @@ func monitorLocalDisksAndHeal(ctx context.Context, objAPI ObjectLayer) {
 						printEndpointError(endpoint, err, true)
 						continue
 					}
+
 					// Calculate the set index where the current endpoint belongs
 					setIndex, _, err := findDiskIndex(z.zones[i].format, format)
 					if err != nil {
@@ -120,12 +152,24 @@ func monitorLocalDisksAndHeal(ctx context.Context, objAPI ObjectLayer) {
 				erasureSetInZoneToHeal[i] = erasureSetToHeal
 			}
 
+			logger.Info("New unformatted drives detected attempting to heal the content...")
+			for i, disks := range localDisksInZoneHeal {
+				for _, disk := range disks {
+					logger.Info("Healing disk '%s' on %s zone", disk, humanize.Ordinal(i+1))
+				}
+			}
+
 			// Heal all erasure sets that need
 			for i, erasureSetToHeal := range erasureSetInZoneToHeal {
 				for _, setIndex := range erasureSetToHeal {
 					err := healErasureSet(ctx, setIndex, z.zones[i].sets[setIndex], z.zones[i].drivesPerSet)
 					if err != nil {
 						logger.LogIf(ctx, err)
+					}
+
+					// Only upon success reduce the counter
+					if err == nil {
+						drivesToHeal--
 					}
 				}
 			}

--- a/cmd/erasure-sets.go
+++ b/cmd/erasure-sets.go
@@ -1414,6 +1414,11 @@ func (s *erasureSets) HealFormat(ctx context.Context, dryRun bool) (res madmin.H
 			return madmin.HealResultItem{}, err
 		}
 
+		refFormat, err = getFormatErasureInQuorum(tmpNewFormats)
+		if err != nil {
+			return madmin.HealResultItem{}, err
+		}
+
 		// kill the monitoring loop such that we stop writing
 		// to indicate that we will re-initialize everything
 		// with new format.

--- a/cmd/erasure-zones.go
+++ b/cmd/erasure-zones.go
@@ -2032,6 +2032,7 @@ type HealthOptions struct {
 // was queried
 type HealthResult struct {
 	Healthy       bool
+	HealingDrives int
 	ZoneID, SetID int
 	WriteQuorum   int
 }
@@ -2086,8 +2087,23 @@ func (z *erasureZones) Health(ctx context.Context, opts HealthOptions) HealthRes
 			}
 		}
 	}
+
+	// check if local disks are being healed, if they are being healed
+	// we need to tell healthy status as 'false' so that this server
+	// is not taken down for maintenance
+	aggHealStateResult, err := getAggregatedBackgroundHealState(ctx, true)
+	if err != nil {
+		logger.LogIf(ctx, fmt.Errorf("Unable to verify global heal status: %w", err))
+		return HealthResult{
+			Healthy: false,
+		}
+	}
+
+	healthy := len(aggHealStateResult.HealDisks) == 0
+
 	return HealthResult{
-		Healthy: true,
+		Healthy:       healthy,
+		HealingDrives: len(aggHealStateResult.HealDisks),
 	}
 }
 

--- a/cmd/global-heal.go
+++ b/cmd/global-heal.go
@@ -73,9 +73,17 @@ func getLocalBackgroundHealStatus() madmin.BgHealState {
 		return madmin.BgHealState{}
 	}
 
+	var healDisks []string
+	for _, eps := range globalBackgroundHealState.getHealLocalDisks() {
+		for _, ep := range eps {
+			healDisks = append(healDisks, ep.String())
+		}
+	}
+
 	return madmin.BgHealState{
 		ScannedItemsCount: bgSeq.getScannedItemsCount(),
 		LastHealActivity:  bgSeq.lastHealActivity,
+		HealDisks:         healDisks,
 		NextHealRound:     UTCNow().Add(durationToNextHealRound(bgSeq.lastHealActivity)),
 	}
 }

--- a/cmd/notification.go
+++ b/cmd/notification.go
@@ -269,21 +269,25 @@ func (sys *NotificationSys) LoadServiceAccount(accessKey string) []NotificationP
 }
 
 // BackgroundHealStatus - returns background heal status of all peers
-func (sys *NotificationSys) BackgroundHealStatus() []madmin.BgHealState {
+func (sys *NotificationSys) BackgroundHealStatus() ([]madmin.BgHealState, []NotificationPeerErr) {
+	ng := WithNPeers(len(sys.peerClients))
 	states := make([]madmin.BgHealState, len(sys.peerClients))
 	for idx, client := range sys.peerClients {
 		if client == nil {
 			continue
 		}
-		st, err := client.BackgroundHealStatus()
-		if err != nil {
-			logger.LogIf(GlobalContext, err)
-		} else {
+		client := client
+		ng.Go(GlobalContext, func() error {
+			st, err := client.BackgroundHealStatus()
+			if err != nil {
+				return err
+			}
 			states[idx] = st
-		}
+			return nil
+		}, idx, *client.host)
 	}
 
-	return states
+	return states, ng.Wait()
 }
 
 // StartProfiling - start profiling on remote peers, by initiating a remote RPC.

--- a/cmd/xl-storage.go
+++ b/cmd/xl-storage.go
@@ -2103,9 +2103,6 @@ func (s *xlStorage) RenameData(srcVolume, srcPath, dataDir, dstVolume, dstPath s
 		legacyDataPath := pathJoin(dstVolumeDir, dstPath, legacyDataDir)
 		// legacy data dir means its old content, honor system umask.
 		if err = os.MkdirAll(legacyDataPath, 0777); err != nil {
-			if isSysErrIO(err) {
-				return errFaultyDisk
-			}
 			return osErrToFileErr(err)
 		}
 
@@ -2116,14 +2113,11 @@ func (s *xlStorage) RenameData(srcVolume, srcPath, dataDir, dstVolume, dstPath s
 
 		for _, entry := range entries {
 			// Skip xl.meta renames further, also ignore any directories such as `legacyDataDir`
-			if entry == xlStorageFormatFile || strings.HasPrefix(entry, SlashSeparator) {
+			if entry == xlStorageFormatFile || strings.HasSuffix(entry, slashSeparator) {
 				continue
 			}
 
 			if err = os.Rename(pathJoin(currentDataPath, entry), pathJoin(legacyDataPath, entry)); err != nil {
-				if isSysErrIO(err) {
-					return errFaultyDisk
-				}
 				return osErrToFileErr(err)
 			}
 		}
@@ -2159,20 +2153,14 @@ func (s *xlStorage) RenameData(srcVolume, srcPath, dataDir, dstVolume, dstPath s
 	}
 
 	if err = renameAll(srcFilePath, dstFilePath); err != nil {
-		if isSysErrIO(err) {
-			return errFaultyDisk
-		}
-		return err
+		return osErrToFileErr(err)
 	}
 
 	if srcDataPath != "" {
 		removeAll(oldDstDataPath)
 		removeAll(dstDataPath)
 		if err = renameAll(srcDataPath, dstDataPath); err != nil {
-			if isSysErrIO(err) {
-				return errFaultyDisk
-			}
-			return err
+			return osErrToFileErr(err)
 		}
 	}
 
@@ -2265,10 +2253,7 @@ func (s *xlStorage) RenameFile(srcVolume, srcPath, dstVolume, dstPath string) (e
 	}
 
 	if err = renameAll(srcFilePath, dstFilePath); err != nil {
-		if isSysErrIO(err) {
-			return errFaultyDisk
-		}
-		return err
+		return osErrToFileErr(err)
 	}
 
 	// Remove parent dir of the source file if empty

--- a/pkg/madmin/heal-commands.go
+++ b/pkg/madmin/heal-commands.go
@@ -294,6 +294,7 @@ type BgHealState struct {
 	ScannedItemsCount int64
 	LastHealActivity  time.Time
 	NextHealRound     time.Time
+	HealDisks         []string
 }
 
 // BackgroundHealStatus returns the background heal status of the

